### PR TITLE
Allow project owners to post updates

### LINKED ISSUE #13: Allow project owners to post updates

### PR DESCRIPTION
**Pull request by abstractcoder**
Work in progress. 

- [x] Create update
- [x] View update
- [x]  Edit update
- [x] Delete update
- [x] Add tests
- [x] Implement permission checks in Post controller
- [x] Fix failing tests

Closes #13

### CODE REVIEW

File: lib/digital_public_works/posts.ex [hunk 1]
```diff
@@ -51,10 +57,11 @@ defmodule DigitalPublicWorks.Posts do
       {:error, %Ecto.Changeset{}}
 
   """
```
Comment (abstractcoder) [inline lib/digital_public_works/posts.ex]: @nickgartmann what is a good way to handle passing multiple associated structs to a context method? Both my current solution, and the previous solution of passing positional arguments feels a bit wrong

File: lib/digital_public_works/posts.ex [hunk 2]
```diff
@@ -51,10 +57,11 @@ defmodule DigitalPublicWorks.Posts do
...
-  def create_post(%Project{} = project, attrs \\ %{}) do
+  def create_post(%{"project" => project, "user" => user} = attrs \\ %{}) do
```
Comment (abstractcoder) [inline lib/digital_public_works/posts.ex]: @nickgartmann what is a good way to handle passing multiple associated structs to a context method? Both my current solution, and the previous solution of passing positional arguments feels a bit wrong

Comment (jeregrine) [reply lib/digital_public_works/posts.ex]:
```
attrs = Map.merge(attrs, %{ "project_id" => project.id, "user_id" => user.id})
```

File: lib/digital_public_works/posts/post.ex [hunk 3]
```diff
@@ -3,21 +3,24 @@ defmodule DigitalPublicWorks.Posts.Post do
...
 
     timestamps()
   end
 
   @doc false
   def changeset(post, attrs) do
     post
-    |> cast(attrs, [:body])
-    |> validate_required([:body])
+    |> cast(attrs, [:title, :body])
```
Comment (jeregrine) [inline lib/digital_public_works/posts/post.ex]:
```
post
|> cast(attrs, [:title, :body, :user_id, :post_id])
|> validate_required([:title, :body, :user_id, :post_id])
``

Comment (jeregrine) [reply lib/digital_public_works/posts/post.ex]: Also add a foreign key validation here https://hexdocs.pm/ecto/Ecto.Changeset.html#foreign_key_constraint/3

File: lib/digital_public_works_web/plugs/auth.ex [hunk 4]
```diff
@@ -9,13 +9,17 @@ defmodule DigitalPublicWorks.Plugs.Auth do
     if conn.assigns[:current_user] do
       conn
     else
-      current_user = if user_id = get_session(conn, :current_user_id) do
-        Accounts.get_user!(user_id)
-      end
+      current_user =
+        if user_id = get_session(conn, :current_user_id) do
+          try do
+            Accounts.get_user!(user_id)
```
Comment (jeregrine) [inline lib/digital_public_works_web/plugs/auth.ex:15]: Add an `Accounts.get_user(user_id)` to the accounts context that returns {:ok, user} or {:error, err} and use a case.

**Review by nickgartmann — APPROVED** [review]

**Pull request merged**